crates/chain/src/spk_client.rs

```diff
@@ -204,10 +240,35 @@ impl<K: Ord + Clone> FullScanRequest<K> {
     pub fn from_chain_tip(chain_tip: CheckPoint) -> Self {
         Self {
             chain_tip,
+            tx_cache: TxCache::new(),
             spks_by_keychain: BTreeMap::new(),
         }
     }
 
+    /// Add to the [`Transaction`] cache that allows the chain source to avoid re-fetching full
+    /// transactions.
```

Comment:
I think it's enough to state the purpose of `TxCache` (to avoid re-fetching) once in the documentation for the type itself, rather than every time it makes an appearance, lol.

```suggestion
    /// Add to the [`Transaction`] cache held by the request.
```

Reply:
Fixed.